libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2599,6 +2599,22 @@ bool AP_Mission::calc_rewind_pos(Mission_Command& rewind_cmd)
     return true;
 }
 
+// get first navigation command which also has valid location, taking into account resuming mission
+bool AP_Mission::get_starting_nav_cmd_with_loc(Mission_Command& cmd) {
+    uint16_t cmd_index = _restart ? AP_MISSION_CMD_INDEX_NONE : _nav_cmd.index;
+    if (cmd_index == AP_MISSION_CMD_INDEX_NONE) {
+        cmd_index = AP_MISSION_FIRST_REAL_COMMAND;
+    }
+    while (get_next_nav_cmd(cmd_index, cmd)) {
```

Comment:
can this watchdog with a do_jump and a loiter_time with 0,0 lat/lon

Reply:
The fix for this would be to change the while loop to a for loop and loop for some some small ish number, there are examples of this in mission somewhere I think.

Reply:
would be good to double check there are no other pre-existing cases of this issue too

Reply:
I went ahead with @IamPete1's suggestion, similar to `is_takeoff_next`. It has the downside that some complex missions may wrongly pass the check.